src/fullfile.c

```diff
@@ -153,6 +153,25 @@ int download_fullfiles(struct list *files, int *num_downloads)
 		free_string(&targetfile);
 	}
 
+	/* remove tar duplicates from the list */
+	need_download = list_sort(need_download, file_sort_hash);
```

Comment:
Sorting a list is very expensive. We prevent a download to be performed twice by using the hashmap in curl_async. 
Do we need to filter files in this stage?

Reply:
yes, otherwise those files that are not actually going to be downloaded because use the same tar are going to be counted when we determine the total download size, then when we are reporting download progress we are never going to reach 100%